src/ports/postgres/modules/deep_learning/madlib_keras_automl.py_in

```diff
@@ -835,10 +835,10 @@ class AutoMLHyperopt(KerasAutoML):
             if param_value_list[-1] == 'linear':
                 return hp.uniform(cp, param_value_list[0], param_value_list[1])
             elif param_value_list[-1] == 'log':
-                return hp.loguniform(cp, param_value_list[0], param_value_list[1])
+                return hp.loguniform(cp, np.log(param_value_list[0]), np.log(param_value_list[1]))
             else:
                 plpy.error("DL: Please choose a valid distribution type for '{0}': {1}".format(
-                    cp, ['linear', 'log']))
+                    cp[:-2], ['linear', 'log']))
```

Comment:
Would this work for `momentum_15`  or `momentum_111` etc.?

Reply:
Good point, though a rare occurrence. I've made a general purpose function to deal with all of these cases.